admin/server/organizations.go

```diff
@@ -301,63 +314,66 @@ func (s *Server) UpdateBillingSubscription(ctx context.Context, req *adminv1.Upd
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	planChange := true
 	for _, sub := range subs {
 		if sub.Plan.ID == plan.ID {
-			planChange = false
-			break
+			return nil, status.Errorf(codes.FailedPrecondition, "organization already subscribed to the plan %s", plan.Name)
 		}
 	}
 
-	if planChange {
-		c, err := s.admin.PaymentProvider.FindCustomer(ctx, org.PaymentCustomerID)
+	// plan change needed
+	// check for valid payment method
+	c, err := s.admin.PaymentProvider.FindCustomer(ctx, org.PaymentCustomerID)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	if !c.HasPaymentMethod {
+		return nil, status.Errorf(codes.FailedPrecondition, "no valid payment method found for the organization")
+	}
+
+	// don't allow plan downgrades, only superuser can downgrade
+	if planDowngrade(plan, org) && !claims.Superuser(ctx) {
+		return nil, status.Errorf(codes.FailedPrecondition, "plan downgrade not allowed")
+	}
```

Comment:
Why do we not allow downgrades? What if someone doesn't use a bigger plan and wants to switch to a smaller one?

Reply:
@nishantmonu51 suggested we should not allow downgrades, for this they should contact us. Downgrades can be tricky if they are already over some quotas.

Reply:
I wonder if it would be better to let people downgrade, and then in the beginning just have a job that checks/alerts if they don't get within the lower quotas within a day or so. Then we can just contact or hibernate non-compliant projects without causing trouble for people who want to downgrade for legitimate reasons.

Reply:
removed the restriction, just logging in case of downgrade

Reply:
Yes, we intentionally wanted to keep the scope limited here and not allow downgrade, as downgrade would require checking if the quotas for the lower plan are met and only allow to downgrade. The expected flow here for a user would be to create a support ticket, FE team will manually verify the quotas and then downgrade directly in orb. 

Other reasons  for not allowing downgrade is for enterprise users who have signed a multi-year contract and we don't want them to downgrade unless explicitly approved. Since its not going to be a common case and in many such cases we want to have manual intervention. We do not allow downgrade at present.